src/frontend/js/confirmation-popups.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * This file contains all functions which are needed for toggling the confirmation popup
+ */
+type EventHandler = (event: Event) => any;
+const handlers = new Map<HTMLElement, EventHandler>();
+
+window.addEventListener("load", () => {
+  // event handler for showing confirmation popups
+  document
+    .querySelectorAll(".confirmation-button")
+    .forEach((el) => el.addEventListener("click", showConfirmationPopup));
+  // event handler for closing confirmation popups
+  document
+    .querySelectorAll("#confirmation-dialog button")
+    .forEach((el) => el.addEventListener("click", closeConfirmationPopup));
+});
+
+export function showConfirmationPopup(event: Event) {
+  event.preventDefault();
+  const button = (event.target as HTMLElement).closest("button");
+  // Set confirmation data
+  document.getElementById(
+    "confirmation-subject"
+  ).innerHTML = button.getAttribute("data-confirmation-subject");
+  document.getElementById("confirmation-title").innerHTML = button.getAttribute(
+    "data-confirmation-title"
+  );
+  document.getElementById("confirmation-text").innerHTML = button.getAttribute(
+    "data-confirmation-text"
+  );
+  const confirmationPopup = document.getElementById("confirmation-dialog");
+  confirmationPopup
+    .querySelector("form")
+    .setAttribute("action", button.getAttribute("data-action"));
```

Comment:
You can't write beyond the attribute with setAttribute. In this case you could however write `javascript:someJavascriptCodeThatDoesBadStuff()` and it would execute. But this is coming sanitized from the backend, right?

Reply:
Yes, it's only used with the `{% url %}` template-tag... so I'll mark the warning as false positive...